hexbytes/_utils.py

```diff
@@ -11,27 +11,26 @@ def to_bytes(val: Union[bool, bytearray, bytes, int, str]) -> bytes:
     Convert a hex string, integer, or bool, to a bytes representation.
     Alternatively, pass through bytes or bytearray as a bytes value.
     """
-    if isinstance(val, bytes):
-        return val
-    elif isinstance(val, str):
-        return hexstr_to_bytes(val)
-    elif isinstance(val, bytearray):
-        return bytes(val)
-    elif isinstance(val, bool):
-        return b"\x01" if val else b"\x00"
-    elif isinstance(val, int):
-        # Note that this int check must come after the bool check, because
-        #   isinstance(True, int) is True
-        if val < 0:
-            raise ValueError(f"Cannot convert negative integer {val} to bytes")
+    type_var = type(val)
+    if type_var is bytes:
+        return val  # type: ignore
```

Comment:
My concern with this approach is that it doesn't handle subclasses. It looks like `HexBytes(Hexbytes(0))` would raise a `TypeError`, for example

Reply:
Hm maybe the code is good enough as-is :+1: